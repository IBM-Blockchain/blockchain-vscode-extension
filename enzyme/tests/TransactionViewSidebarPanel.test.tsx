import React  from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SidebarPanel from '../../src/components/TransactionViewSidebarPanel/TransactionViewSidebarPanel';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('SidebarPanel component', () => {

    let mySandBox: sinon.SinonSandbox;

    beforeEach(async () => {
        mySandBox = sinon.createSandbox();
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot when panelType = \`buttons\`', async () => {
        const component: any = renderer
            .create(<SidebarPanel panelType={'buttons'}></SidebarPanel>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the expected snapshot when panelType = \`filters\`', async () => {
        const component: any = renderer
            .create(<SidebarPanel panelType={'filters'}></SidebarPanel>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the expected snapshot when panelType = \`log\`', async () => {
        const component: any = renderer
            .create(<SidebarPanel panelType={'log'}></SidebarPanel>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('will one day allow a user to create a new transaction', async() => {
        const createTxnSpy: sinon.SinonSpy = sinon.spy(SidebarPanel.prototype, 'createTxn');
        const component: any = mount(<SidebarPanel panelType='buttons'></SidebarPanel>);
        component.find('button[id=\'create-button\']').simulate('click');
        createTxnSpy.should.have.been.called;
    });
    
    it('will one day allow a user to create a new transaction', async() => {
        const importTxnSpy: sinon.SinonSpy = sinon.spy(SidebarPanel.prototype, 'importTxn');
        const component: any = mount(<SidebarPanel panelType='buttons'></SidebarPanel>);
        component.find('button[id=\'import-button\']').simulate('click');
        importTxnSpy.should.have.been.called;
    });

});
