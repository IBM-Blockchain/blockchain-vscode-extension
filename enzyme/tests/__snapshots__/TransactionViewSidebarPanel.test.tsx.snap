// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarPanel component should render the expected snapshot when panelType = \`buttons\` 1`] = `
<div
  className="panel-container"
>
  <button
    className="bx--btn bx--btn--field bx--btn--primary"
    disabled={false}
    id="create-button"
    onClick={[Function]}
    tabIndex={0}
    type="button"
  >
    Create a new transaction
  </button>
  <button
    className="bx--btn bx--btn--field bx--btn--primary"
    disabled={false}
    id="import-button"
    onClick={[Function]}
    tabIndex={0}
    type="button"
  >
    Import existing transaction
  </button>
</div>
`;

exports[`SidebarPanel component should render the expected snapshot when panelType = \`filters\` 1`] = `
<div
  className="panel-container"
>
  <h5>
    Filters
  </h5>
</div>
`;

exports[`SidebarPanel component should render the expected snapshot when panelType = \`log\` 1`] = `
<div
  className="panel-container"
>
  <h5>
    Transactions
  </h5>
</div>
`;
